ui/src/main/java/org/pentaho/di/ui/trans/dialog/TransExecutionConfigurationDialog.java

```diff
@@ -144,6 +145,10 @@ private void getVariablesData() {
         tableItem.setText( 2, Const.NVL( variableValue, "" ) );
       }
     }
+    //Internal Varibles should get disabled based on the kettle varible HIDE_INTERNAL_VARIABLES=Y
+    if ( ValueMetaString.convertStringToBoolean( System.getProperty( "HIDE_INTERNAL_VARIABLES", "Y" ) ) ) {
```

Comment:
We should add this string as a constant in the kettle-core Const class and reference it from there vs. having this string in multiple classes.

Reply:
updated